site/src/layout/header/Header.tsx

```diff
@@ -1,21 +1,31 @@
 "use client";
+import { Typography } from "@src/common/components/Typography";
 import { SvgUse } from "@src/common/helpers/SvgUse";
 import { PageLink } from "@src/layout/header/PageLink";
 import { PageLayout } from "@src/layout/PageLayout";
+import { DisableBodyScrolling } from "@src/util/DisableBodyScrolling";
 import { useEffect, useState } from "react";
-import { useIntl } from "react-intl";
-import styled from "styled-components";
+import { FormattedMessage, useIntl } from "react-intl";
+import styled, { css } from "styled-components";
 
 import { GQLHeaderFragment } from "./Header.fragment.generated";
 
 interface Props {
     header: GQLHeaderFragment[];
 }
 
+const headerHeight = 100;
+
 const Header = ({ header }: Props) => {
     const intl = useIntl();
+    const [isMenuOpen, setIsMenuOpen] = useState<boolean>(false);
     const [expandedSubLevelNavigation, setExpandedSubLevelNavigation] = useState<string | null>(null);
 
+    const toggleMenu = () => {
+        if (isMenuOpen) setExpandedSubLevelNavigation(null);
+        setIsMenuOpen((prev) => !prev);
```

Comment:
```suggestion
        if (isMenuOpen) {
            setExpandedSubLevelNavigation(null);
            setIsMenuOpen(false);
        } else {
            setIsMenuOpen(true);    
        }
```

IMO this is more readable